open-metadata-implementation/access-services/subject-area/subject-area-server/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/handlers/SubjectAreaGlossaryHandler.java

```diff
@@ -78,12 +78,15 @@ public SubjectAreaGlossaryHandler(OMRSAPIHelper oMRSAPIHelper) {
         try {
             InputValidator.validateNodeType(className, methodName, suppliedGlossary.getNodeType(), NodeType.Glossary, NodeType.Taxonomy, NodeType.TaxonomyAndCanonicalGlossary, NodeType.CanonicalGlossary);
             final String suppliedGlossaryName = suppliedGlossary.getName();
-
+            final String suppliedGlossaryQualifiedName = suppliedGlossary.getQualifiedName();
             // need to check we have a name
             if (suppliedGlossaryName == null || suppliedGlossaryName.equals("")) {
                 ExceptionMessageDefinition messageDefinition = SubjectAreaErrorCode.GLOSSARY_CREATE_WITHOUT_NAME.getMessageDefinition();
                 throw new InvalidParameterException(messageDefinition, className, methodName, "name");
             } else {
+                if (suppliedGlossaryQualifiedName == null || suppliedGlossaryQualifiedName.equals("")) {
```

Comment:
We can move these constructions to `setUniqueQualifiedName` method and write this:
```
    protected void setUniqueQualifiedNameIfBlank(Node node) {
               if (StringUtils.isBlank(node.getQualifiedName())) {
                    node.setQualifiedName(node.getName() + "." + UUID.randomUUID().toString());
                }
    }
```

And then remove duplicate code.

StringUtils.isBlank specification:
```
 StringUtils.isBlank(null)      = true
 StringUtils.isBlank("")        = true  
 StringUtils.isBlank(" ")       = true  
 StringUtils.isBlank("bob")     = false 
```

Reply:
And I prefer to use "@" instead of "."

Reply:
@immmus I assume you are thinking of the Spring StringUtils, this module should not have a Spring dependancy, so I have followed your feedback but wrote out the tests without the utility.

Reply:
@davidradl I am talking about apache commons StringUtils.

Reply:
> @davidradl I am talking about apache commons StringUtils.

@immmus Thanks for the clarification, I have coded it in a simple if, I think this should be sufficient for now.

Reply:
@davidradl yes, it's sufficient)